agent-launcher/src/com/thoughtworks/go/agent/launcher/AgentLauncherImpl.java

```diff
@@ -57,6 +58,7 @@ public AgentLauncherImpl(AgentProcessParentRunner agentProcessParentCreator) {
 
     public int launch(AgentLaunchDescriptor descriptor) {
         Thread shutdownHook = null;
+        new SystemEnvironment().initializeAgentLogDir();
```

Comment:
We probably need do this in `AgentBootstrapper#main` as well, just before `LogConfigurator.initialize()` as well.